### PR TITLE
Add updatecli subcommand to generate json schema

### LINKED ISSUE #524: Feature Request: Add jsonschema to validate updatecli configuration
### Is your feature request related to a problem?

It would be useful to have a JSON schema to validate updatecli configuration from an IDE.
More information on https://json-schema.org/specification.html

### Solution you'd like

I would like to easily write updatecli configuration from VScode

### Alternatives you've considered

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by olblak**
# Add updatecli subcommand to generate json schema

- [x] Require https://github.com/updatecli/updatecli/issues/444

PR related to https://github.com/updatecli/updatecli/pull/540 but the current one focus on generating json schema while #540 test updatecli manifest with json schema

Fix #524 

The current pullrequest introduces several requirements

1. To retrieve field description, a comment must be one line maximum and me precede the field such
```
// Source defines parameters to retrieve an information from a specific location
type Source struct {
    // Title defines a source title
    title string // This is not a valid comment anymore
} 
``` 
2. Configuration key must be lowercase to be valid, while we could use another convention, we can't mix them. 
```
--- 
    sources:
        default: 
            scmID: # not valid
            scmid: # valid  
```

## Test

To test this pullrequest, you can: 

1. Ensure that test in package `github.com/pkg/core/jsonschema` are passing.
```shell
make test
```

2. You can try to generate an Updatecli jsonschema using the new jsonschema argument 
```shell
go build -o bin/updatecli && ./bin/updatecli jsonschema --directory schema
```

3. Finally, you can configure your IDE such as vscode to match files like ["**/updatecli.di/**/*.yaml","**/updatecli.di/**/*.yml","**/updatecli.di/**/*.tpl"] with the jsonschema file that you generated locally such as "$HOME/Projects/Updatecli/updatecli/schema/config.json" as explained on [vscode documentation](https://code.visualstudio.com/Docs/languages/json#_file-match-syntax)  

### Tradeoff

/

### Potential improvement

This pullrequest won't be usable as is as we need additional cleanup.
.1 Ensure that every field expected lowercase value. for example this [tag](https://github.com/updatecli/updatecli/blob/bb4c8c94120bccab45a20549382721f51a129b25/pkg/core/pipeline/resource/main.go#L31) must be removed

### CODE REVIEW

**Review by dduportal — DISMISSED** [review]
👏

File: cmd/jsonschema.go [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+			}
+		},
+	}
+	jsonschemaDirectory string
+	jsonschemaBaseID    string
+)
+
+func init() {
+	jsonschemaCmd.Flags().StringVarP(&jsonschemaDirectory, "directory", "d", "./", "Export schema to directory")
+	jsonschemaCmd.Flags().StringVarP(&jsonschemaBaseID, "baseID", "b", "https://www.updatecli.io/latest/schema", "Define schema baseID")
```
Comment (lemeurherve) [inline cmd/jsonschema.go:30]: WDYT about "baseid"?

Comment (olblak) [reply cmd/jsonschema.go:30]: I was thinking of the same considering that we are leaning toward lowercase everywhere.
I'll update that

Comment (olblak) [reply cmd/jsonschema.go]: Done

File: pkg/core/pipeline/resource/main.go [hunk 2]
```diff
@@ -37,12 +37,20 @@ type ResourceConfig struct {
...
 }
 
 // New returns a newly initialized Resource or an error
 func New(rs ResourceConfig) (resource Resource, err error) {
-	switch strings.ToLower(rs.Kind) {
+
+	kind := strings.ToLower(rs.Kind)
+
+	if _, ok := GetResourceMapping()[kind]; !ok {
+		return nil, fmt.Errorf("⚠ Don't support resource kind: %v", rs.Kind)
```
Comment (lemeurherve) [inline pkg/core/pipeline/resource/main.go:50]: Hardcoded warning sign

**Pull request merged**